core/capabilities/webapi/outgoing_connector_handler.go

```diff
@@ -153,6 +144,82 @@ func (c *OutgoingConnectorHandler) HandleSingleNodeRequest(ctx context.Context,
 	}
 }
 
+// AwaitConnection attempts to establish a connection to an available gateway.  It iterates through available gateways
+// using the gatewaySelector, attempting to connect to each.  The method respects the provided context, allowing for
+// cancellation or timeout.
+func (c *OutgoingConnectorHandler) AwaitConnection(ctx context.Context) (string, error) {
+	return c.awaitConnectionUntilCanceled(ctx)
```

Comment:
@MStreet3 🤔 Why not just remove awaitConnectionUntilCancelled entirely and collapso the two?

Reply:
Will do, was initially because I had two different implementations.